test/test_limited_api.py

```diff
@@ -0,0 +1,50 @@
+import textwrap
+
+from . import test_projects, utils
+
+limited_api_project = test_projects.new_c_project(
+    setup_py_add=textwrap.dedent(
+        r"""
+        cmdclass = {}
+        extension_kwargs = {}
+        if sys.version_info[:2] >= (3, 8):
+            from wheel.bdist_wheel import bdist_wheel as _bdist_wheel
+
+            class bdist_wheel_abi3(_bdist_wheel):
+                def finalize_options(self):
+                    _bdist_wheel.finalize_options(self)
+                    self.root_is_pure = False
+
+                def get_tag(self):
+                    python, abi, plat = _bdist_wheel.get_tag(self)
+                    return python, "abi3", plat
+
+            cmdclass["bdist_wheel"] = bdist_wheel_abi3
+            extension_kwargs["define_macros"] = [("Py_LIMITED_API", "0x03080000")]
+            extension_kwargs["py_limited_api"] = True
+    """
+    ),
+    setup_py_extension_args_add="**extension_kwargs",
+    setup_py_setup_args_add="cmdclass=cmdclass",
+)
```

Comment:
I suppose this is the way to get abi3 wheel from `pip wheel`/`python -m build`? Because we can't pass `--py-limited-api`? I wonder if there's a cleaner way to do it. I suppose we'd want to document this too.

Reply:
A follow-up on this question - should a reference version of this bdist_wheel specify in the tag that it's a "3.8-ABI" wheel, rather than using whatever it's being built by?

Reply:
Most packages will hard code this into setup.py. A package either produces abi3 wheels or regular wheels, usually they don't select between the two, I think. So you'd hardcode in the define in the source code, and `py_limited_api=True` is passed to `setup()`.

Reply:
I tested something a bit more complicated than what most packages will do. As @henryiii mentioned, and I think so to, "a package either produces abi3 wheels or regular wheels, usually they don't select between the two" and it'll be hardcoded in the source code.
Even for some packages that still support cp27 or some that depends on an API being made stable in a version more recent than the minimum supported, the logic will probably hardcoded.
I really don't think it's cibuildwheel's place to document how to build an `abi3` wheel. Only if workarounds were to arise, specific to cibuildwheel, those should be documented then.

Reply:
> I tested something a bit more complicated than what most packages will do.

Understood.

> I really don't think it's cibuildwheel's place to document how to build an abi3 wheel. Only if workarounds were to arise, specific to cibuildwheel, those should be documented then.

The only thing that might be considered specific is the inability to pass `--py-limited-api` to bdist_wheel - most docs online still reference this. But since that calling method is deprecated, it does indeed make sense to document a more modern way across the ecosystem. Ideally, indeed, in packaging.python.org. Then we can link it from our docs.